### PR TITLE
Set cq token

### PR DESCRIPTION
**Pull request by teaguetomesh**
Contains shared code used by `qiskit-superstaq` and `cirq-superstaq` to let a user update their CQ token. 

Client side PRs coming shortly...

### CODE REVIEW

**Comment by teaguetomesh** [comment]
Client support for server changes made in SuperstaQ [PR2051](https://github.com/SupertechLabs/SuperstaQ/pull/2051)

File: general_superstaq/user_config_test.py [hunk 1]
```diff
@@ -80,6 +80,20 @@ def test_ibmq_set_token(  # pylint: disable=missing-function-docstring
     assert service.ibmq_set_token("valid token") == "Your IBMQ account token has been updated"
 
 
+@mock.patch(
+    "general_superstaq.superstaq_client._SuperstaQClient.cq_set_token",
```
Comment (richrines1) [inline general_superstaq/user_config_test.py:84]: should also work and mocks (slightly) less:
```suggestion
    "general_superstaq.superstaq_client._SuperstaQClient.post_request",
```

File: general_superstaq/user_config_test.py [hunk 2]
```diff
@@ -80,6 +80,20 @@ def test_ibmq_set_token(  # pylint: disable=missing-function-docstring
...
+    return_value="Your CQ account token has been updated",
+)
+def test_mq_set_token(  # pylint: disable=missing-function-docstring
```
Comment (richrines1) [inline general_superstaq/user_config_test.py:87]: typo
```suggestion
def test_cq_set_token(  # pylint: disable=missing-function-docstring
```

File: general_superstaq/user_config_test.py [hunk 3]
```diff
@@ -80,6 +80,20 @@ def test_ibmq_set_token(  # pylint: disable=missing-function-docstring
...
+    mock_cq: mock.MagicMock,
```
Comment (richrines1) [inline general_superstaq/user_config_test.py:88]: (assuming prior suggestion)
```suggestion
    mock_post_request: mock.MagicMock,
```

File: general_superstaq/superstaq_client.py [hunk 4]
```diff
@@ -304,6 +304,19 @@ def ibmq_set_token(  # pylint: disable=missing-param-doc,differing-param-doc
 
         return self.post_request("/ibmq_token", json_dict)
 
+    def cq_set_token(  # pylint: disable=missing-param-doc,differing-param-doc
+        self, json_dict: Dict[str, str]
+    ) -> str:
+        """Makes a POST request to SuperstaQ API to set CQ token field in database.
+
+        Args:
+            cq_token: dictionary with CQ token string entry.
```
Comment (richrines1) [inline general_superstaq/superstaq_client.py:313]:
```suggestion
            json_dict: Dictionary with CQ token string entry.
```

File: general_superstaq/superstaq_client.py [hunk 5]
```diff
@@ -304,6 +304,19 @@ def ibmq_set_token(  # pylint: disable=missing-param-doc,differing-param-doc
 
         return self.post_request("/ibmq_token", json_dict)
 
+    def cq_set_token(  # pylint: disable=missing-param-doc,differing-param-doc
```
Comment (richrines1) [inline general_superstaq/superstaq_client.py:307]: are these pragmas necessary? if not:
```suggestion
    def cq_set_token(
```

File: general_superstaq/user_config_test.py (hunk 1 above)
Comment (teaguetomesh) [reply general_superstaq/user_config_test.py:84]: In this case should we also change the mock for the `ibmq_set_token` test?

File: general_superstaq/superstaq_client.py (hunk 4 above)
Comment (teaguetomesh) [reply general_superstaq/superstaq_client.py:313]: Same as above -- I mostly copied this code from what already existed for the `ibmq_token`s, should I also update the naming for those functions for consistency?

File: general_superstaq/user_config_test.py (hunk 1 above)
Comment (richrines1) [reply general_superstaq/user_config_test.py:84]: ah hadn't noticed that, yeah that'd be great!

**Review by richrines1 — APPROVED** [review]
lgtm!

**Pull request merged**